buildSrc/build.gradle.kts

```diff
@@ -1,24 +1,23 @@
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 plugins {
-    `java-gradle-plugin`
     `kotlin-dsl`
+    `kotlin-dsl-precompiled-script-plugins`
 }
 
 repositories {
     google()
     mavenCentral()
 }
 
+// NOTE: when updating any of these keep in sync with the version catalog
 dependencies {
-    compileOnly(gradleApi())
+    implementation(gradleApi())
```

Comment:
Can we fail the SDK build if the versions we have here isn't in sync with the very catalog?

Reply:
I'm not sure of an easy way to achieve this. I've asked Daniel whether he has any suggestions on how to use version catalog with precompiled build script plugin dependencies, so we can see what he says & alter accordingly.

Reply:
The `buildSrc` projects are a special (legacy) flavour of an included build. For historical reasons, they do not require a `settings.gradle` file, but (I believe [1]) you can add one without issue. From the settings file, you can load the catalogue from the `../gradle/libs.versions.toml`.

[1] If adding the settings.gradle file fails for some reason; just convert the `buildSrc` project into a proper include build. I don't think you need to do that.